kstatemachine/src/commonMain/kotlin/ru/nsk/kstatemachine/IState.kt

```diff
@@ -13,6 +13,7 @@ import kotlin.reflect.KClass
 @StateMachineDslMarker
 interface IState : TransitionStateApi, VisitorAcceptor {
     val name: String?
+    val displayName: String?
```

Comment:
I see the problem here, that it is not scaling. If in future there will be some other kind of meta information that should be attached to state/transitionadditional filed will be necessary again. 

I suggest  to add 
sealed interface MetaInfo
interface StateMetaInfo : MetaInfo
interface TransitionMetaInfo : MetaInfo

then add  interface for your specific need

interface DisplayMetaInfo : StateMetaInfo {
    val displayName: String
}

export code should check for metaState filead if it is instance of StateMetaInfo.

now you can have implementatino of this interface.

Reply:
Yes, it should be a part of the library. 
So MetaInfo is a marker for extendability. And DisplayNameMetaInfo is a standard (defined by the library) extension.

Reply:
do you see it as metaInfo parameter in state and transition?
```
IState(
    metaInfo: DisplayNameStateMetaInfo
)
```
Or custom state/transition can implement MetaState interface if there is need in it?
```
CustomState : IState, DisplayNameStateMetaInfo {
   displayName: String
}
```

Reply:
Second way will be problematic because all extension functions (create state/transition) do not know about extra contract

Reply:
I see it almost like first variant.

```
// the state/transition is a host for meta info. It just stores it as an optional field
IState {
     name: String? // current name
     metaInfo: MetaInfo? // new property
}

// there might be many different MetaInfo sub-classes (interfaces)
// our case
interface DisplayNameStateMetaInfo: MetaInfo {
     val displayName: String
}
// or
interface NoteMetaInfo: MetaInfo {
     val note: String
}
// or
interface BannerMetaInfo: MetaInfo {
     val someComplexData: ComplexData
}

// when a user constructs a State he can combine those sub-interfaces into one concrete class/object.
DefaultState(
     name = "state",
     metaInfo = object: NoteMetaInfo, DisplayNameStateMetaInfo { 
            val displayName = "Customized name"
            val note = "Very long text"
     }
)

// MetaInfo consumers/readers (like ExportPlantUmlVisitor) can get metaInfo property from a state/transition and try to use it by casting to expected type.

val displayName = (state.metaInfo as? DisplayNameStateMetaInfo?).displayName
displayName?.let {
    // do something 
}

// this way, as I see, the library provides extendable api for attaching any kind of meta info to its primitives.
```

Reply:
Exactly what i did already. Will push soon

Reply:
pushed